mysql-test/suite/rpl/t/rpl_semi_sync_shutdown_await_ack.inc

```diff
@@ -142,12 +200,11 @@ EOF
 --enable_reconnect
 --source include/wait_until_connected_again.inc
 
---eval SET @@GLOBAL.debug_dbug= "$sav_master_dbug"
-let $status_var= Rpl_semi_sync_master_clients;
-let $status_var_value= 0;
-source include/wait_for_status_var.inc;
---eval SET @@GLOBAL.rpl_semi_sync_master_enabled = 0
-show status like 'Rpl_semi_sync_master_status';
+# Because we restarted, we have to store the default values again for later
+# restoration at cleanup
+set @sav_master_timeout= @@global.rpl_semi_sync_master_timeout;
+set @sav_enabled_master= @@GLOBAL.rpl_semi_sync_master_enabled;
+set @sav_master_dbug= @@GLOBAL.debug_dbug;
```

Comment:
In this case (@user variables lost due to server restart), I think it's better to simply use mysqltest --let $save_xxx variables that preserve their value, and simply use --disable-query-log to avoid a possibly changing default value to get into the .result file.

Or even simpler, if the server restarts, there's no need to restore the values, the restart already resets all variables to default. If the values are changed again after the restart, it's sufficient to save just after restart, avoiding the issue mentioned in the comment here.

Reply:
Ah yeah, disable query log indeed takes care of the awkwardness. Thanks!